metrics/core/src/main/java/tech/pegasys/pantheon/metrics/MetricCategory.java

```diff
@@ -12,45 +12,9 @@
  */
 package tech.pegasys.pantheon.metrics;
 
-import java.util.EnumSet;
-import java.util.Set;
+public interface MetricCategory {
 
-public enum MetricCategory {
-  BIG_QUEUE("big_queue"),
-  BLOCKCHAIN("blockchain"),
-  EXECUTORS("executors"),
-  JVM("jvm", false),
-  NETWORK("network"),
-  PEERS("peers"),
-  PROCESS("process", false),
-  PERMISSIONING("permissioning"),
-  KVSTORE_ROCKSDB("rocksdb"),
-  KVSTORE_ROCKSDB_STATS("rocksdb", false),
-  RPC("rpc"),
-  SYNCHRONIZER("synchronizer"),
-  TRANSACTION_POOL("transaction_pool");
+  String getName();
 
-  // Why not BIG_QUEUE and ROCKSDB?  They hurt performance under load.
-  public static final Set<MetricCategory> DEFAULT_METRIC_CATEGORIES =
-      EnumSet.complementOf(EnumSet.of(BIG_QUEUE, KVSTORE_ROCKSDB, KVSTORE_ROCKSDB_STATS));
-
-  private final String name;
-  private final boolean pantheonSpecific;
-
-  MetricCategory(final String name) {
-    this(name, true);
-  }
-
-  MetricCategory(final String name, final boolean pantheonSpecific) {
-    this.name = name;
-    this.pantheonSpecific = pantheonSpecific;
-  }
-
-  public String getName() {
-    return name;
-  }
-
-  public boolean isPantheonSpecific() {
-    return pantheonSpecific;
-  }
+  boolean isApplicationSpecific();
```

Comment:
Instead of having a `boolean` here + the `applicationPrefix` configured on the `MetricsSystem`, what about just storing an `Optional<String> applicationPrefix` on `MetricsCategory`?

Reply:
I was skeptical at first but have come around.  This does simplify things and given an application would typically have one enum for its categories it actually works out pretty nicely.